docs/DeveloperGuide.md

```diff
@@ -1,25 +1,168 @@
+that is entered. 
+* A verification check is added to the add command of the version 2.0 of WhereGotTime.
+
+E.g.
+* add /CS2113 Lec /Mon /1200-1400 /LT21<br/>
+* add /CS2113 Lec /Mon /1300-1400 /LT22<br/>
+* add /CS2040C Tut /Mon /1300-1500 /LT32<br/>
+
+The above would result in the first line being add,
+while the other two will result in the app giving an error message
+prompting the user to re-enter again.<br/>
+
+### Edit Function<br/>
+Class Diagram
+
+![](team/ClassDiagram_EditCommand.png)<br/>
+
+### Delete Function<br/>
+
+### Clear Function<br/>
+
+### Display Function<br/>
+
+### Compare Function<br/>
+#### System Architecture of the Compare Function<br/>
+![](team/CompareCommand_ClassDiagram.jpg) <br/>
+The System Architecture given above explains the high-level design of the <b>COMPARE</b> command.<br/>
+
+Listed below are some features we will be highlighting:<br/>
+
+* Ui: The User Interface of the app<br/>
+* CompareCommand: The main logic command of the <b>COMPARE</b> function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* User: User(s) of the program<br/>
+* UserList: An arraylist of User<br/>
+
+#### Design of the Compare Function<br/>
+![](team/CompareCommand_UML_Diagram.jpg)<br/>
+
+1) The <b>COMPARE</b> function gets the inputs from the User and retrieves the User's / Target User's<br/>
+timetables based on a specific day. It will then compare the timetables and return an arraylist
+of common timeslots.<br/>
+2)  It will then pass the arraylist onto Ui class to display the range of common <br/>
+timeslots the User and Target User have on a specific day of the week.<br/>
+
+
+#### Sequence Design of Compare Function<br/>
+![](team/CompareCommand_SequenceDesign.jpg) <br/>
+
+#### Improvements to be made to Version 2.0<br/>
+* Display a range of timings instead of individual hours<br/>
+* Remove edge cases<br/>
+
+E.g.<br/>
+* compare /Alex /Mon<br/>
+* compare /Tammy /Mon<br/>
+
+These cases above would no longer output "0000HR, 0100HR" etc but
+instead "0000HR - 1000HR"
+
+### Login Function
+#### System Architecture of the Login Function<br/>
+![](team/LogInCommand_Architecture_Diagram_v001.png)
+* UI: The User Interface of the app<br/>
+* LogInCommand: The main logic command of the Login function<br/>
+* Parser: The main logic that takes input data and builds the command.
+* User: The Class where username, password, etc are stored
+* UserList: The Class that creates an ArrayList where the User objects will be stored
+
+#### Class Diagram
+![](team/LogInCommand_Class_Diagram_v002.png)
+1.) The LogInCommand checks the UserList to check if the inputted user already exists.
+<br/>
+2.) In the case that the User is a first time user, the LogInCommand will create a new User object then add it into 
+the UserList. 
+
+#### Sequence Diagram
+![](team/LogInCommand_Sequence_v002.png)
+
```

Comment:
Missing colons for all objects
![image](https://user-images.githubusercontent.com/60027323/97514844-b07cf700-19ca-11eb-8a6f-e80988714f82.png)

Reply:
Opt should be used instead of alt here as there are no alternatives provided.